pkg/operator/hostendpointscontroller/host_endpoints_controller.go

```diff
@@ -112,52 +113,51 @@ func (c *HostEndpointsController) sync() error {
+	required := hostEndpointsAsset()
+
 	discoveryDomain, err := c.getEtcdDiscoveryDomain()
 	if err != nil {
 		return fmt.Errorf("unable to determine etcd discovery domain: %v", err)
 	}
 
-	// list etcd member pods
-	pods, err := c.podLister.List(labels.Set{"k8s-app": "etcd"}.AsSelector())
+	if required.Annotations == nil {
+		required.Annotations = map[string]string{}
+	}
+	required.Annotations["alpha.installer.openshift.io/dns-suffix"] = discoveryDomain
 
-	// get dns names of ready etcd member pods
-	var addresses []string
-	for _, pod := range pods {
-		var ready bool
-		for _, condition := range pod.Status.Conditions {
-			if condition.Type == corev1.PodReady {
-				ready = condition.Status == corev1.ConditionTrue
+	// create endpoint addresses for each node
+	nodes, err := c.nodeLister.List(labels.Set{"node-role.kubernetes.io/master": ""}.AsSelector())
+	if err != nil {
+		fmt.Errorf("unable to list expected etcd member nodes: %v", err)
+	}
+	endpointAddresses := &required.Subsets[0].Addresses
+	for _, node := range nodes {
+		var nodeInternalIP string
+		for _, nodeAddress := range node.Status.Addresses {
+			if nodeAddress.Type == corev1.NodeInternalIP {
+				nodeInternalIP = nodeAddress.Address
 				break
 			}
 		}
-		if ready {
-			dnsName, err := c.getEtcdDNSName(discoveryDomain, pod.Status.HostIP)
-			if err != nil {
-				return fmt.Errorf("unable to determine dns name for etcd member on node %s: %v", pod.Spec.NodeName, err)
-			}
-			addresses = append(addresses, dnsName)
+		if len(nodeInternalIP) == 0 {
+			return fmt.Errorf("unable to determine internal ip address for node %s", node.Name)
+		}
+		dnsName, err := c.getEtcdDNSName(discoveryDomain, nodeInternalIP)
+		if err != nil {
+			return fmt.Errorf("unable to determine etcd member dns name for node %s: %v", node.Name, err)
 		}
-	}
-
-	if len(addresses) == 0 {
-		return fmt.Errorf("no etcd member pods are ready")
-	}
-
-	required := hostEndpointsAsset()
-
-	if required.Annotations == nil {
-		required.Annotations = map[string]string{}
-	}
-	required.Annotations["alpha.installer.openshift.io/dns-suffix"] = discoveryDomain
 
-	sort.Strings(addresses)
-	for i, address := range addresses {
-		required.Subsets[0].Addresses = append(required.Subsets[0].Addresses, corev1.EndpointAddress{
-			Hostname: strings.TrimSuffix(address, "."+discoveryDomain),
-			IP:       net.IPv4(byte(192), byte(0), byte(2), byte(i+1)).String(),
+		*endpointAddresses = append(*endpointAddresses, corev1.EndpointAddress{
+			IP:       nodeInternalIP,
+			Hostname: strings.TrimSuffix(dnsName, "."+discoveryDomain),
+			NodeName: &node.Name,
 		})
 	}
 
+	sort.Slice(*endpointAddresses, func(i, j int) bool {
+		return (*endpointAddresses)[i].Hostname < (*endpointAddresses)[j].Hostname
+	})
+
 	// if etcd-bootstrap exists, keep it (at the end of the list)
 	existing, err := c.endpointsLister.Endpoints("openshift-etcd").Get("host-etcd")
 	if err != nil && !errors.IsNotFound(err) {
```

Comment:
update this to the switch/case style from https://github.com/openshift/cluster-etcd-operator/pull/112

Reply:
> update this to the switch/case style from #112

Lack of this results in stomping out the the bootstrap etcd and we fail to start.